react/features/prejoin/components/Prejoin.js

```diff
@@ -265,12 +268,16 @@ class Prejoin extends Component<Props, State> {
         const { _closeDialog, _onDropdownClose, _onOptionsClick, _setName, _showDialog } = this;
         const { showJoinByPhoneButtons } = this.state;
 
+        // We need the check for interfaceConfig because isButtonEnabled will work only if interfaceConfig is defined.
+        const showConferenceInfo
```

Comment:
Could you pls add this to `mapStateToProps` and pass it as a separate prop? Looks good otherwise.

Reply:
Can you elaborate why do you want me to do this?

Reply:
1.It's easier to reason about components when you have distinct props. (If someone who has never seen this component looks at it it's easier to understand it, imo).

2. If we remove `interfaceConfig` from the global scope, `mapStateToProps` would be the place where the prop would be computed.

3. Ideally this would be a separate selector which could be memoized. (but having it in `mapStateToProps` I think is good enough for now, as I haven't seen selector memoization).

Reply:
Agree with 1!

I disagree with 2. IMHO when and if we remove interfaceConfig from the global scope we can go and move things in mapStateToProps if needed. But I don't see any point of trying to overcomplicate things for the sole purpose that when we do a hypothetical change it will be easier for us. For example - we talk about moving interfaceConfig from global scope since 3 years maybe :)  

3 is a good idea!

I will move it to mapStateToProps because I agree with 1!


Thanks for the explanation!

Reply:
Thanks!